### PR TITLE
unify external command invocation

### LINKED ISSUE #33: Unify external command invocation
utils.py uses several different approaches to invoke an external command:

* [subprocess.Popen](https://github.com/packit-service/ogr/blob/master/ogr/utils.py#L103)
* [subprocess.check_call](https://github.com/packit-service/ogr/blob/master/ogr/utils.py#L122)
* [subprocess.check_output](https://github.com/packit-service/ogr/blob/master/ogr/utils.py#L171)

The recommended approach (since Python 3.5) is to use [subprocess.run](https://docs.python.org/3/library/subprocess.html#subprocess.run).

Also, there's lots of external libraries for this use case:
* http://amoffat.github.io/sh/
* https://www.pyinvoke.org/
* https://plumbum.readthedocs.io
* https://pexpect.readthedocs.io

### PR DESCRIPTION
**Pull request by lbarcziova**
Fixes #33

### CODE REVIEW

File: ogr/utils.py [hunk 1]
```diff
@@ -42,16 +41,12 @@ def clone_repo_and_cd_inside(repo_name, repo_ssh_url, namespace):
...
-        logger.debug(
-            "Clone exited with {} and output: {}".format(proc.returncode, output)
-        )
-        if "does not exist yet" not in output:
-            break
-        sleep(1)
-    else:
+    proc = subprocess.run(
+        ["git", "clone", repo_ssh_url], stderr=subprocess.PIPE, timeout=CLONE_TIMEOUT
+    )
```
Comment (lachmanfrantisek) [inline ogr/utils.py]: :+1:

File: ogr/utils.py [hunk 2]
```diff
@@ -42,16 +41,12 @@ def clone_repo_and_cd_inside(repo_name, repo_ssh_url, namespace):
...
+    output = proc.stderr.read().decode()
+    logger.debug("Clone exited with {} and output: {}".format(proc.returncode, output))
+    if "does not exist yet" in output:
```
Comment (jpopelka) [inline ogr/utils.py]: I don't see the `yet` in the failed `git clone` output

File: ogr/utils.py [hunk 3]
```diff
@@ -42,16 +41,12 @@ def clone_repo_and_cd_inside(repo_name, repo_ssh_url, namespace):
...
-            "Clone exited with {} and output: {}".format(proc.returncode, output)
-        )
-        if "does not exist yet" not in output:
-            break
-        sleep(1)
-    else:
+    proc = subprocess.run(
+        ["git", "clone", repo_ssh_url], stderr=subprocess.PIPE, timeout=CLONE_TIMEOUT
+    )
+    output = proc.stderr.read().decode()
```
Comment (jpopelka) [inline ogr/utils.py]:
```suggestion
    output = proc.stderr.decode()
```

since `AttributeError: 'bytes' object has no attribute 'read'`

**Comment by lachmanfrantisek** [comment]
@jpopelka sorry for the quick merge...

@lbarcziova Can you please fix the suggestions by @jpopelka in the second pull-request?

File: ogr/utils.py (hunk 3 above)
Comment (lbarcziova) [reply ogr/utils.py]: I will fix it

**Pull request merged**